File: galleries/tiny/base_r/stacked_bar.R
library(tibble)

df <- tribble(
  ~Contestant, ~Fruit, ~`Number Eaten`,
  "Alex", "Apples", 2,
  "Alex", "Oranges", 1,
  "Alex", "Bananas", 3,
  "Jordan", "Apples", 1,
  "Jordan", "Oranges", 3,
  "Jordan", "Bananas", 2
)

data <- with(df, tapply(`Number Eaten`, list(Contestant, Fruit), FUN = sum))
barplot(data, col = c("darkblue", "red"), legend = TRUE)

File: galleries/tiny/observable/stacked_bar.mjs
import * as Plot from "@observablehq/plot";

export async function chart() {
  const data = [
    { Fruit: "Apples", Contestant: "Alex", "Number Eaten": 2 },
    { Fruit: "Oranges", Contestant: "Alex", "Number Eaten": 1 },
    { Fruit: "Bananas", Contestant: "Alex", "Number Eaten": 3 },
    { Fruit: "Apples", Contestant: "Jordan", "Number Eaten": 1 },
    { Fruit: "Oranges", Contestant: "Jordan", "Number Eaten": 3 },
    { Fruit: "Bananas", Contestant: "Jordan", "Number Eaten": 2 },
  ];

  return Plot.auto(data, {
    x: "Fruit",
    y: "Number Eaten",
    mark: "bar",
    color: "Contestant",
  }).plot({ color: { legend: true } });
}
